app/src/main/kotlin/de/markusfisch/android/binaryeye/actions/wifi/WifiNetworkInfoFactory.kt

```diff
@@ -0,0 +1,41 @@
+package de.markusfisch.android.binaryeye.actions.wifi
+
+object WifiNetworkFactory {
+    private val wifiRegex = Regex("^WIFI:(T:(WEP|WPA|nopass))?;S:(.*);(P:(.*))?;(H?);$")
+    private val hexRegex = Regex("^[0-9a-f]*$", RegexOption.IGNORE_CASE)
+
+    fun parse(data: String): WifiNetworkInfo? {
+        return wifiRegex.matchEntire(data)?.groupValues?.let {
+            val security = it[2].let {
+                if (it == "" || it == "nopass") return null
+                it
+            }
+
+            val ssid = quoteUnlessHex(it[3])
+            if (ssid.isEmpty()) return null
+
+            val password = if (it[5] == "") null else quoteUnlessHex(it[5])
+
+            val hidden = it[6] == "H"
+            return WifiNetworkInfo(
```

Comment:
wikipedia states:
>Special characters `"` (quotation mark), `;` (semicolon), `,` (comma) and `:` (colon) should be escaped with a backslash (`\`) as in MECARD encoding. For example, if an SSID were `"foo;bar\baz"`, with quotation marks part of the literal SSID name itself, this would be encoded as: `WIFI:S:\"foo\;bar\\baz\";;`

Your parsing would fail to parse this as it doesn't remove the escaping. This should be done before creating the `WifiNetworkInfo`.